language/stdlib/modules/libra_configs.move

```diff
@@ -1,32 +1,101 @@
 address 0x0:
 module LibraConfig {
     use 0x0::Transaction;
-    use 0x0::LibraSystem;
+    use 0x0::LibraAccount;
+    use 0x0::LibraTimestamp;
 
     // A generic singleton resource that holds a value of a specific type.
     resource struct T<Config: copyable> { payload: Config }
 
-    // Set a config item to a new value and trigger a reconfiguration.
-    // This can only be invoked by the Association address
-    public fun set<Config: copyable>(payload: Config) acquires T {
+    struct NewEpochEvent {
+        epoch: u64,
+    }
+
+    resource struct Configuration {
+        epoch: u64,
+        last_reconfiguration_time: u64,
+        events: LibraAccount::EventHandle<NewEpochEvent>,
+    }
+
+    // This can only be invoked by the Association address, and only a single time.
+    // Currently, it is invoked in the genesis transaction
+    public fun initialize_configuration() {
+        let sender = Transaction::sender();
+        // Only callable by the Association address for now.
+        Transaction::assert(sender == 0xA550C18, 1);
+
+        move_to_sender<Configuration>(Configuration {
+            epoch: 0,
+            last_reconfiguration_time: 0,
+            events: LibraAccount::new_event_handle<NewEpochEvent>(),
+        });
+    }
+
+    // We can't return reference here, this returns a copy of the config value.
+    public fun get<Config: copyable>(): Config acquires T {
         let sender = Transaction::sender();
 
         // Only callable by the Association address for now.
         Transaction::assert(sender == 0xA550C18, 1);
 
+        Transaction::assert(::exists<T<Config>>(sender), 24);
+        let t_ref = borrow_global_mut<T<Config>>(sender);
```

Comment:
This should be a lint error, but you're borrowing a mut and only using it as read-only